components/XElement/XElement.astro

```diff
@@ -168,6 +172,41 @@ for (const name in attrs) {
 	}
 }
 
+const Payload: string = `
+	(async function($$,$){$=this;$$=(await import($$.src)).default;${
+			defineVarsString ?? ``
+		}${
+			listenerString ?? ``
+		}${
+			listenerOfOnce ?? ``
+		}}).call(this.previousSibling,this.remove()||this)
+`
+/** Transforms the JS/TS input using ESBuild */
+const transformer = async (code:string):Promise<string>=> (await transformWithEsbuild(code,`xelement.ts`,{minify: true})).code 
+
+/** Create a hashmap to store the transformed code */
+const hash = Object.create(null)
+
+/** Return transformed code, remembering previously transformed code. */
+const memo = async (code: string) => {
+  //NoOp if the code is undefined, 
+  if (code === undefined) return
```

Comment:
It doesn’t appear that `memo` is ever used with an undefined value, since `Payload` is always `string`.

```suggestion
```

Reply:
It was a strange bug that came up with some elements where they have no functions associated to them and kept coming up undefined, it then would break the build process causing the server to break, this sidestepped that,

Reply:
Would you be able to share a replication of this?

On [Line 206](https://github.com/aFuzzyBear/astro-ui/pull/28/files#diff-104773d26925fc29e25c6660ff3881b0900eda497e69aa78ee9fa12d9328cb77R206), we see that `memo` is used as `await memo(Payload)`, and on [Line 175](https://github.com/aFuzzyBear/astro-ui/pull/28/files#diff-104773d26925fc29e25c6660ff3881b0900eda497e69aa78ee9fa12d9328cb77R175) we see that `Payload` cannot be `undefined`.

Reply:
When I had placed a `console.log('this would breaks')` in place of return in the no-op, this is what was logged to the console, allowing the rest of the XElements to be processed without interruption

![image](https://user-images.githubusercontent.com/28299972/144769296-40b6822b-fdd1-4d50-bb8b-80b36396f90f.png)